bin/utils/component-utils.sh

```diff
@@ -503,3 +503,187 @@ process_component_appfw_plugin() {
+    echo ${desktopIframe_id}
+    desktopIframe_index=`expr $desktopIframe_index + 1`
+    desktopIframe_id=$(read_component_manifest "${component_dir}" ".desktopIframePlugins[${desktopIframe_index}].id" 2>/dev/null)
+  done
+  return 0
+}
+
+###############################
+# Verifies a specific component by performing a fast check. If the component
+# has any services, it will check if the status of those services are "UP". If
+# it has any desktop plugins it will check if it exists on the zowe instance.
+#
+# Note: this function calls is dependent on various utility functions and
+#       environment variables. Simply source bin/internal/prepare-environment.sh
+#       to import the required dependencies.
+#
+# Required environment variables:
+# - NODE_HOME
+# - ROOT_DIR
+# - ZOWE_EXPLORER_HOST
+# - DISCOVERY_PORT
+# - ZOWE_ZLUX_SERVER_HTTPS_PORT
+#
+# Example:
+# - Verify service and desktop plugin of explorer-jes component
+#   i.e: verify_component_instance "${ROOT_DIR}/components/explorer-jes"
+#
+# @param string   component directory
+verify_component_instance() {
+  component_id=$1
+  rc_failures=0
+
+  component_dir=$(find_component_directory "${component_id}")
+
+  service_ids=$(list_component_service_id "${component_dir}")
+
+  print_and_log_message "=========================================="
+
+  for service_id in $service_ids; do
+    json_response=$(node "${ROOT_DIR}"/bin/utils/curl.js https://"${ZOWE_EXPLORER_HOST}":"${DISCOVERY_PORT}"/eureka/apps/"${service_id}" -k -H 'Accept: application/json' -J 2>/dev/null)
+    status_index=0
+    service_status=$(echo "${json_response}" | read_json - .application.instance[${status_index}].status 2>/dev/null)
+    while [[ -n ${service_status} ]]; do
+      if [[ "${service_status}" == "UP" ]]; then
+        print_and_log_message "- service ${service_id} is registered successfully and status is: ${service_status}"
+      else
+      # This case is currently used but will be implmented for future purposes
+        print_and_log_error_message "- service ${service_id} is registered but is currently ${service_status}"
+        rc_failures=`expr $rc_failures + 1`
+      fi
+      status_index=`expr $status_index + 1`
+      service_status=$(echo "${json_response}" | read_json - .application.instance[${status_index}].status 2>/dev/null)
+    done
+    if [[ ${status_index} -eq 0 ]]; then
+        print_and_log_error_message "- service ${service_id} is not registered properly!"
+        rc_failures=`expr $rc_failures + 1`
+    fi
+  done
+
+  if [ -z "${service_ids}" ]; then
+     print_and_log_message "- No services exist for this component"
+  fi
+  
+  desktop_ids=$(list_component_plugin_id "${component_dir}")
+
+  for desktop_id in $desktop_ids; do
+    desktop_identifier=$(node "${ROOT_DIR}"/bin/utils/curl.js https://"${ZOWE_EXPLORER_HOST}":"${ZOWE_ZLUX_SERVER_HTTPS_PORT}"/plugins -k | read_json - .pluginDefinitions[].identifier)
```

Comment:
can we add a line `log_message "${component_id} desktop plugin ${desktop_id} identifier: ${desktop_identifier}"` after this line for debugging purpose.